crates/admin/src/schema_registry/updater.rs

```diff
@@ -0,0 +1,938 @@
+// Copyright (c) 2024 - Restate Software, Inc., Restate GmbH.
+// All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use crate::schema_registry::error::{
+    ComponentError, DeploymentError, SchemaError, SubscriptionError,
+};
+use http::{HeaderValue, Uri};
+use restate_schema::component::{ComponentLocation, ComponentSchemas, HandlerSchemas};
+use restate_schema::deployment::DeploymentSchemas;
+use restate_schema::SchemaInformation;
+use restate_schema_api::component::{ComponentType, HandlerType};
+use restate_schema_api::deployment::DeploymentMetadata;
+use restate_schema_api::invocation_target::{
+    InputRules, InputValidationRule, InvocationTargetMetadata, OutputContentTypeRule, OutputRules,
+};
+use restate_schema_api::subscription::{
+    EventReceiverComponentType, Sink, Source, Subscription, SubscriptionValidator,
+};
+use restate_service_protocol::discovery::schema;
+use restate_types::identifiers::{DeploymentId, SubscriptionId};
+use serde::{Deserialize, Serialize};
+use std::collections::hash_map::Entry;
+use std::collections::HashMap;
+use tracing::{info, warn};
+
+/// Responsible for updating the provided [`SchemaInformation`] with new
+/// schema information. It makes sure that the version of schema information
+/// is incremented on changes.
+#[derive(Debug, Default)]
+pub(crate) struct SchemaUpdater {
```

Comment:
Nit: All `pub(crate)` instances here are potentially just noise. Unless `mod updater` is public, it's simpler to just keep everything here public.

Reply:
True. Will remove it.